tests/build_test.ts

```diff
@@ -0,0 +1,89 @@
+import * as path from "$std/path/mod.ts";
+import { puppeteer } from "./deps.ts";
+import { assert } from "$std/_util/asserts.ts";
+import { startFreshServer, waitForText } from "$fresh/tests/test_utils.ts";
+
+Deno.test("build snapshot and restore from it", async (t) => {
+  const fixture = path.join(Deno.cwd(), "tests", "fixture_build");
+  const outDir = path.join(fixture, "_fresh");
+
+  try {
+    await t.step("build snapshot", async () => {
+      await new Deno.Command(Deno.execPath(), {
+        args: [
+          "run",
+          "-A",
+          path.join(fixture, "dev.ts"),
+          "build",
+        ],
+        stdin: "null",
+        stdout: "inherit",
+        stderr: "inherit",
+      }).output();
+
+      assert((await Deno.stat(outDir)).isDirectory, "Missing output directory");
+    });
+
+    await t.step("check snapshot file", async () => {
+      const snapshot = JSON.parse(
+        await Deno.readTextFile(path.join(outDir, "snapshot.json")),
+      );
+      assert(
+        Array.isArray(snapshot["island-counter_default.js"]),
+        "Island output file not found in snapshot",
+      );
+      assert(
+        Array.isArray(snapshot["main.js"]),
+        "main.js output file not found in snapshot",
+      );
+      assert(
+        Array.isArray(snapshot["signals.js"]),
+        "signals.js output file not found in snapshot",
+      );
+      assert(
+        Array.isArray(snapshot["deserializer.js"]),
+        "deserializer.js output file not found in snapshot",
+      );
+    });
+
+    await t.step("restore from snapshot", async () => {
+      const { lines, serverProcess, address, output } = await startFreshServer({
+        args: [
+          "run",
+          "--allow-env",
+          "--allow-read",
+          "--allow-net=0.0.0.0",
```

Comment:
Is this a safe `--allow-net`? I have bad memories around `0.0.0.0`, not sure if they apply here.

Reply:
I'm admittedly not sure why deno needs this. Normally `127.0.0.1` or `localhost` should suffice, but when I use these I get the following error:

```txt
error: Uncaught (in promise) PermissionDenied: Requires net access to "0.0.0.0:8000", run again with the --allow-net flag
    await Deno.serve(opts, handler).finished;
```

Reply:
Interesting. Seems this works fine then, I'll keep an eye on it for the future.